File: sidecars/synaplan-compute/tests/hostile/symlink_out.py
# expected-result: succeeded
# reason:
# note: Symlink from /out to /etc/passwd is not followed on artefact pull; list omits it
"""symlink_out.py — plant a symlink in /out that must not be pulled."""
import os
import sys

os.makedirs("/out", exist_ok=True)
target = "/out/passwd"
try:
    os.symlink("/etc/passwd", target)
except FileExistsError:
    pass
# Also write a real artefact so the run still produces a regular file.
with open("/out/ok.txt", "w", encoding="utf-8") as f:
    f.write("ok\n")
sys.exit(0)

File: sidecars/synaplan-compute/tests/hostile/node/symlink_out.js
// expected-result: succeeded
// reason:
// note: Symlink from /out to /etc/passwd is not followed on artefact pull
// symlink_out.js — plant a symlink in /out that must not be pulled.
const fs = require('fs');
try {
  fs.symlinkSync('/etc/passwd', '/out/passwd');
} catch (e) {
  // The link may already exist from a previous attempt; that is fine.
}
fs.writeFileSync('/out/ok.txt', 'ok\n');
